javascript/src/components/inputs/InputDropdown.jsx

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+var React = require('react/addons');
+var Input = require('react-bootstrap').Input;
+var Spinner = require('../common/Spinner');
+var Col = require('react-bootstrap').Col;
+
+var InputDropdown = React.createClass({
+    mixins: [React.addons.LinkedStateMixin],
+    PLACEHOLDER: "placeholder",
+    getInitialState() {
+        return {
+            selectedInput: this.PLACEHOLDER
+        };
+    },
+    _formatInput(input) {
+        return <option key={input.id} value={input.id}>{input.title} ({input.type})</option>;
+    },
+    _sortByTitle(input1, input2) {
+        return input1.title.localeCompare(input2.title);
+    },
+    _onClick() {
+        this.props.onClick(this.state.selectedInput);
+    },
+    render() {
+        if (this.props.inputs) {
+            var inputs = this.props.inputs.sort(this._sortByTitle).map(this._formatInput);
+            return (
+                <Col md={12}>
+                    <Col md={6}>
+                        <Input type='select' valueLink={this.linkState('selectedInput')} placeholder={this.PLACEHOLDER}>
+                            <option value={this.PLACEHOLDER}>--- Select an Input ---</option>
+                            {inputs}
```

Comment:
This is generating a warning in the browser's JS console:
`Warning: Using Maps as children is not yet fully supported. It is an experimental feature that might be removed. Convert it to a sequence / iterable of keyed ReactElements instead.`